internal/changestream/eventmultiplexer/eventmultiplexer.go

```diff
@@ -178,48 +173,36 @@ func (e *EventMultiplexer) Report() map[string]any {
 	}
 }
 
-func (e *EventMultiplexer) unsubscribe(subscriptionID uint64) {
-	select {
-	case <-e.catacomb.Dying():
-		return
-	case <-e.stream.Dying():
-		return
-	case e.unsubscriptionCh <- subscriptionID:
-	}
-
-	// Decrease number of subscriptions metric.
-	e.metrics.SubscriptionsDec()
-}
-
 func (e *EventMultiplexer) loop() error {
 	ctx, cancel := e.scopedContext()
 	defer cancel()
 
 	defer func() {
 		for _, sub := range e.subscriptions {
-			sub.close()
+			sub.Kill()
 		}
 		e.subscriptions = nil
 		e.subscriptionsByNS = nil
+		e.subscriptionsAll = nil
 	}()
 
 	for {
+		e.cleanupDeadSubscriptions(ctx)
```

Comment:
I wonder if you just do this after a dispatch?

If the tomb or stream is dying, then it will be take care of via the defer above.